components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/impl/WorkersApiServiceImpl.java

```diff
@@ -0,0 +1,904 @@
+            siddhiAppMetricsHistory.setMemory(memory);
+            List<List<Object>> throughput = metricsDBHandler.selectAppOverallMetrics("throughput",
+                    carbonId, timeInterval, appName, System.currentTimeMillis());
+            siddhiAppMetricsHistory.setThroughput(throughput);
+            List<List<Object>> latency = metricsDBHandler.selectAppOverallMetrics("latency",
+                    carbonId, timeInterval, appName, System.currentTimeMillis());
+            siddhiAppMetricsHistory.setLatency(latency);
+            siddhiAppList.add(siddhiAppMetricsHistory);
+            String jsonString = new Gson().toJson(siddhiAppList);
+            return Response.ok().entity(jsonString).build();
+        } else {
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+    }
+
+
+    /**
+     * This method return the both siddi apptext view and flow chart.PS: Currently implemetented till text view.
+     *
+     * @param id      workerid of the siddhi app
+     * @param appName siddhiapp name
+     * @return the responce with the text view of the siddhi app.
+     * @throws NotFoundException
+     */
+    @Override
+    public Response getSiddhiAppDetails(String id, String appName) throws NotFoundException {
+        String[] hostPort = id.split(WORKER_KEY_GENERATOR);
+        if (hostPort.length == 2) {
+            InmemoryAuthenticationConfig usernamePasswordConfig = workerInmemoryConfigs.get(id);
+            if (usernamePasswordConfig == null) {
+                usernamePasswordConfig = getAuthConfig(id);
+            }
+            String workerURIBody = generateURLHostPort(hostPort[0], hostPort[1]);
+            feign.Response workerSiddiActiveApps = WorkerServiceFactory.getWorkerClient(PROTOCOL + workerURIBody,
+                    usernamePasswordConfig.getUserName(), usernamePasswordConfig.getPassWord()).getSiddhiApp(appName);
+            String responseAppBody = workerSiddiActiveApps.body().toString();
+            return Response.ok().entity(responseAppBody).build();
+        }
+        return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+
+    /**
+     * Try to reach the worker node;
+     *
+     * @param workerURI host:port
+     * @return response from the worker.
+     */
+    private String getWorkerGeneralDetails(String workerURI, String workerId) {
+        InmemoryAuthenticationConfig usernamePasswordConfig = workerInmemoryConfigs.get(workerId);
+        if (usernamePasswordConfig == null) {
+            usernamePasswordConfig = getAuthConfig(workerId);
+        }
+        try {
+            feign.Response workerResponse = WorkerServiceFactory.getWorkerClient(PROTOCOL + workerURI,
+                    usernamePasswordConfig.getUserName(), usernamePasswordConfig.getPassWord()).getSystemDetails();
+            return workerResponse.body().toString();
+        } catch (feign.RetryableException e) {
+            logger.warn(workerId + " Worker not reachable.");
```

Comment:
add stacktrace to the log. 
`logger.warn(workerId + " Worker not reachable.", e);`